src/kademlia/dht_storage.cpp

```diff
@@ -453,6 +464,15 @@ namespace
 			touch_item(i->second, addr);
 		}
 
+		int get_infohashes_sample(entry& item) const override
```

Comment:
well, part of this is for historical reasons, but right now it's very convenient due to the way the storage is used in other parts, even performance wise. I would say that a refactor is warranted at later point.